tests/ifs/ifs.c

```diff
@@ -110,25 +179,49 @@ static int scan_init(struct test *test)
                 ifs0 = open(PATH_SYS_IFS_BASE "intel_ifs_0", O_DIRECTORY | O_PATH | O_CLOEXEC);
         }
 
-
+        /* see if driver files exist */
         if (ifs0 < 0 || faccessat(ifs0, "run_test", R_OK, 0) < 0 ||
-                faccessat(ifs0, "status", R_OK, 0) < 0 || faccessat(ifs0, "details", R_OK, 0) < 0) {
+                faccessat(ifs0, "status", R_OK, 0) < 0 ||
+                faccessat(ifs0, "details", R_OK, 0) < 0 ||
+                faccessat(ifs0, "image_version", R_OK, 0) < 0 ||
+                faccessat(ifs0, "current_batch", R_OK, 0) < 0) {
                 log_info("not supported (missing kernel module, firmware, or invalid HW)");
                 close(ifs0);
                 return -EOPNOTSUPP;
         }
```

Comment:
Maybe remove the `faccessat` portions and leave only the check for `ifs0` file descriptor, which case remove the `close(ifs0)` too. Once the kernel API is set in stone, we're not going to see multiple versions of the driver any more.